pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1118,13 +1118,24 @@ func maybeRetryBoot(ctx *domainContext, status *types.DomainStatus) {
 	}
 	defer file.Close()
 
+	if err := hyper.Task(status).VirtualTPMSetup(status.DomainName, agentName, ctx.ps, warningTime, errorTime); err != nil {
+		log.Errorf("Failed to setup virtual TPM for %s: %s", status.DomainName, err)
+		config.VirtualTPM = false
+	} else {
+		config.VirtualTPM = true
+	}
+
 	if err := hyper.Task(status).Setup(*status, *config, ctx.assignableAdapters, nil, file); err != nil {
 		//it is retry, so omit error
 		log.Errorf("Failed to create DomainStatus from %+v: %s",
 			config, err)
 	}
 
-	status.TriedCount += 1
+	if err := hyper.Task(status).VirtualTPMTerminate(status.DomainName); err != nil {
```

Comment:
By the way, why do we need to terminate here?... It's a path of VM activation, as I understand.

Reply:
I guess I wanted to put it somewhere here:
https://github.com/lf-edge/eve/blob/40fccb1ad15fef72934c464aaecd4a52226beece/pkg/pillar/cmd/domainmgr/domainmgr.go#L1143

Reply:
move it to the error path.